tests/mypy_test.py

```diff
@@ -57,6 +57,7 @@
 Platform: TypeAlias = Annotated[str, "Must be one of the entries in SUPPORTED_PLATFORMS"]
 
 
+@dataclass
```

Comment:
Does this really need to be a dataclass? Making something both an argparse Namespace and a dataclass seems confusing.

Reply:
I don't think so? I may have tried both approaches and accidentally left both in >.<

Reply:
Those are all false positives in this case, since `argparse` uses `setattr()` to initialise all of the instance variables on the namespace object passed to it. I suppose pyright could theoretically add special casing to `argparse.Namespace` objects to prevent this error being emitted in this kind of situation, but I'll wager Eric probably won't be enthusiastic about that idea...

Reply:
In that case, I'd argue "meaning it's used like a dataclass". (though I don't doubt I could be wrong). Could do `@dataclass(init=False, repr=False, eq=False)` if we don't need/want extra methods.

Reply:
It's probably simpler just to decorate it with `dataclass(init=False)` and remove the inheritance from `argparse.Namespace` (which isn't necessary for argparse, it just gives us nicer `__repr__` and `__eq__` methods -- if we're using dataclasses, there's no need for the inheritance here).

Reply:
I filed the bug report with pyright as I figured Eric might be interested -- his response indicates that the `reportUninitializedInstanceVariable` diagnostic is basically unmaintained and may be removed altogether at some point `¯\_(ツ)_/¯`